lib/irq.c

```diff
@@ -423,8 +416,8 @@ vfu_irq_message(vfu_ctx_t *vfu_ctx, uint32_t subindex)
     struct vfio_user_irq_info irq_info;
 
     ret = validate_irq_subindex(vfu_ctx, subindex);
-    if (ret < 0) {
-        return -1;
+    if (ret == false) {
```

Comment:
ditto:

if (!validate_irq_subindex()) {

Here, we also shouldn't set ret, as that's an int

Reply:
Done